### ISSUE TITLE
Extensions issue: update / reinstallation

### ISSUE THREAD

**Issue by ArmandPhilippot** [microsoft/vscode#107997]
Hello,

As of today, I have encountered a problem with the extensions of VS Code.

I published my own theme on the marketplace a few months ago (`armandphilippot.coldark`). So far, I had no problems updating it. Today I am doing a new update. Everything is going smoothly. I wait. Unable to update theme in VS Code. The option "_Install another version_" remains grayed out.

I switch to Windows, open VS Code, and the extension updates with no problem. The problem therefore seems to come from the VS Code version on Linux, at least Manjaro which is my distribution.

I tried uninstalling my theme to reinstall it. VS Code cannot find it. It is however available on the marketplace, and the Windows version finds it. I tried to install it through the terminal, the result is the same. I first thought of a corrupted package ... I uninstalled another extension to test. Once uninstalled, I cannot reinstall it. The "_Install_" option does not appear in the open window of the extension (displayed before uninstallation). If I search for the theme again, VS Code cannot find it. In both cases, it was a theme (New Moon by Tania Rascia for the second). So I tested with another type of extension (Preview on Live Server ... I cannot remember the publisher name) and the result is the same.

Here is the information provided by the "_About_" menu in "_Help_":  
```
Version: 1.49.1
Commit: 58bb7b2331731bf72587010e943852e13e6fd3cf
Date: 2020-09-23T22:26:38.399Z
Electron: 9.2.1
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Linux x64 5.4.67-1-MANJARO
```
Steps to Reproduce:

1. Uninstall an extension
2. Search this extension in VS Code
3. Result: "Extension not found"

For step 2, if I replace it with:
* Open a terminal
* `code --install-extension armandphilippot.coldark`
The result is the same: Extension 'armandphilippot.coldark' not found.

Here is the output in the terminal:
![vscode-extension-bug](https://user-images.githubusercontent.com/59021693/94966840-3a39c180-04fe-11eb-829f-a68738a65add.jpg)

If that helps, here's what fits into the console of Developer Tools:
![vscode-console-output](https://user-images.githubusercontent.com/59021693/94967028-9c92c200-04fe-11eb-8af3-7cbcbbd9f856.jpg)

**Also note:** yesterday there were many updates on Manjaro including VS Code. In the pacman logs, I see "1.48.2-1 -> 1.49.1-1" for `code`. The bug must be related to this update since I never had a problem before ...

**Edit:** This problem only happens with an extension that I just uninstalled. Those which have not been installed do offer an "Install" option.

Thanks in advance,

Do not hesitate to ask me for additional information if necessary ... I have tried to be as exhaustive as possible.

Armand.

**Comment by ArmandPhilippot** [comment]
I have performed other tests since yesterday. I don't know if I should edit my post or comment afterwards. The second option seemed better to me in terms of readability.

I also forgot to specify yesterday, my desktop environment is XFCE. I don't know if it can help.

I have completely uninstalled VS Code.
* `pacman -Rsn code`
* `rm -R ~/.vscode-oss`
* `rm -R "~/.config/Code - OSS"`
* `rm -R ~/.config/code-oss`

Then I reinstalled it. Now if I remove an extension that I just installed. I can reinstall it.  
**Note:** I didn't try yesterday with an extension that I just installed, only with extensions that had been installed for a while. So I don't know if this problem is fixed or not ...

Even though this problem seems to be resolved, the extensions I uninstalled yesterday are still not found. Whether it's my theme (Coldark), the "_New Moon_" theme or the "_Preview on Web Server_" extension.

If I open the Developer Tools console, I also have a lot more problems than yesterday. On the other hand, I do not see anything related to the extensions which cannot be found.

Here are two screenshots (same session, the output was too long so it's cut in 2):

|Part 1|Part 2|
|:-:|:-:|
|![vscode-new-console-output1](https://user-images.githubusercontent.com/59021693/94990776-f9ce5800-057e-11eb-8f95-8699556a4f2a.jpg)|![vscode-new-console-output2](https://user-images.githubusercontent.com/59021693/94990784-018dfc80-057f-11eb-9476-98c8196d2918.jpg)|

Hoping that this additional information can help identify the problem ...

Thanks in advance!

**Comment by gschaffner** [comment]
The change causing this was introduced in 1.49.1-1, in which the Arch packagers switched the marketplace to OpenVSX. https://git.archlinux.org/svntogit/community.git/commit/trunk?h=packages/code&id=b62a02ec645f16528c0269faf12e362a55efda75

A temporary workaround for this is to use the prior version: `sudo pacman -U https://archive.archlinux.org/packages/c/code/code-1.49.0-2-x86_64.pkg.tar.zst`

The proper solution for this is to reach out to your extension maintainers and ask them to publish to OpenVSX. See https://www.eclipse.org/community/eclipse_newsletter/2020/march/1.php (Ctrl + F for `missing`).

**Comment by ArmandPhilippot** [comment]
Thanks for these information! I suspected it was related to the last version, but I didn't know where to look to find what had changed.

I did not know OpenVSX ... I see this is what is used by VS Codium. So in the end, both are using the same marketplace now. I understand for VS Codium, a little less for Code - OSS.

Even if I understand the interest, deciding to no longer support Microsoft's marketplace risks limiting the visibility of small open source projects ... I don't think I'm the only one who doesn't know OpenVSX.

I just tested. By posting my theme to OpenVSX, it is now available in Code!

If anyone has the same problem, another temporary workaround: go to the [Marketplace](https://marketplace.visualstudio.com/vscode), search for the extension, then click "_Download extension_". All that remains is to install it in Code using the `vsix` file.

Anyway, it looks like this isn't a "problem" from VS Code itself, but from Archlinux, so I guess I can close this issue. Thanks again.
